core/src/main/java/org/primefaces/extensions/model/inputplace/Place.java

```diff
@@ -0,0 +1,174 @@
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package org.primefaces.extensions.model.inputplace;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+/**
+ * Model object for InputPlace representing a single Google Place location.
+ *
+ * @author Melloware &lt;mellowaredev@gmail.com&gt;
+ * @since 14.0
+ */
+public class Place implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private final String placeId;
+    private final String formattedAddress;
+    private final String addressLine;
+    private final String postalCode;
+    private final String city;
+    private final String state;
+    private final String country;
+    private final String administrativeAreaLevel1;
+    private final String administrativeAreaLevel2;
+    private final String administrativeAreaLevel3;
+    private final String types;
+    private final double latitude;
+    private final double longitude;
+
+    public Place(final String clientId, final Map<String, String> params) {
+        this.placeId = params.get(clientId + "_place_id");
+        this.formattedAddress = params.get(clientId + "_formatted_address");
+        this.addressLine = params.get(clientId + "_address");
+        this.postalCode = params.get(clientId + "_postcode");
+        this.city = params.get(clientId + "_city");
+        this.state = params.get(clientId + "_state");
+        this.country = params.get(clientId + "_country");
+        this.latitude = Double.parseDouble(params.get(clientId + "_lat"));
+        this.longitude = Double.parseDouble(params.get(clientId + "_lng"));
+        this.administrativeAreaLevel1 = params.get(clientId + "_administrative_area_level_1");
+        this.administrativeAreaLevel2 = params.get(clientId + "_administrative_area_level_2");
+        this.administrativeAreaLevel3 = params.get(clientId + "_administrative_area_level_3");
+        this.types = params.get(clientId + "_types");
+    }
+
+    public Place(String placeId, String formattedAddress, String addressLine, String postalCode, String city, String state, String country,
```

Comment:
Wow! Sonar does not like when the amount of parameters exceeds 7, and neither do I 🙈. Maybe we could have a builder instead?

Reply:
Is this even used? (Just did a rough scan). Maybe we an simply delete it. Or make the field non-final.

Reply:
Agreed I was just following your pattern but I will remove the constructor!